state-sync/storage-service/server/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use diem_infallible::RwLock;
+use diem_types::{epoch_change::EpochChangeProof, transaction::TransactionListWithProof};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use storage_interface::DbReaderWriter;
+use storage_service_types::{
+    DataSummary, EpochEndingLedgerInfoRequest, ProtocolMetadata, ServerProtocolVersion,
+    StorageServerSummary, StorageServiceRequest, StorageServiceResponse,
+    TransactionsWithProofRequest,
+};
+use thiserror::Error;
+
+#[cfg(test)]
+mod tests;
+
+// TODO(joshlind): make these configurable.
+/// Storage server constants.
+pub const MAX_TRANSACTION_CHUNK_SIZE: u64 = 1000;
+pub const STORAGE_SERVER_VERSION: u64 = 1;
+
+#[derive(Clone, Debug, Deserialize, Error, PartialEq, Serialize)]
+pub enum Error {
+    #[error("Storage error encountered: {0}")]
+    StorageErrorEncountered(String),
+    #[error("Unexpected error encountered: {0}")]
+    UnexpectedErrorEncountered(String),
+}
+
```

Comment:
add a `pub type Result<T, E = Error> = ::std::result::Result<T, E>;`

Reply:
Can you elaborate? Why would we want this? These errors should be internal to the storage service only.

Reply:
This I think is just more convenience along with the `Error` so you don't have to keep typing `Error`.  It's optional

Reply:
As weirdly as it sounds, I prefer typing `Error` everywhere. Especially since so much of our code uses anyhow (which I'm not a big fan of). This way, I know exactly what Error type I'm getting 😄